github/prci.py

```diff
@@ -19,13 +20,27 @@
 )
 from internals.gql import util, queries
 
-
 logger = logging.getLogger(__name__)
 
 
 ERROR_BACKOFF_TIME = 600
 
 
+def initial_cleanup():
+    """
+        The main idea of this function is to clean stucked vagrant process
+        and remove the vagrant vms before a new run.
+    """
+    stuck_process = "ps aux | grep -i '[v]agrant up' | awk '{ print $2 }' | xargs kill -9"
```

Comment:
`pkill` and `pgrep` aren't goign to work, vagrant is not called by a binary called `vagrant` but using `ruby`:

``` bash
[root@reproducer-3130 ~]# systemctl status prci
● prci.service - FreeIPA PR CI
   Loaded: loaded (/etc/systemd/system/prci.service; enabled; vendor preset: disabled)
   Active: active (running) since Tue 2019-06-18 16:12:12 -03; 4min 28s ago
 Main PID: 2393 (prci.py)
    Tasks: 1 (limit: 4915)
   Memory: 38.7M
   CGroup: /system.slice/prci.service
           └─2393 /usr/bin/python3 /root/freeipa-pr-ci/github/prci.py reproducer-3130 --config /root/.config/freeipa-pr-ci/config.yml

jun 18 16:16:23 reproducer-3130.lab.cerberus.domain bash[2393]: INFO: __main__: Skipping fedora-29/test_webui_users of #6: waiting for dependencies
jun 18 16:16:23 reproducer-3130.lab.cerberus.domain bash[2393]: INFO: __main__: Skipping fedora-29/customized_ds_config_install of #6: waiting for dependencies
jun 18 16:16:23 reproducer-3130.lab.cerberus.domain bash[2393]: INFO: __main__: Skipping fedora-29/dns_locations of #6: not enough resources
jun 18 16:16:23 reproducer-3130.lab.cerberus.domain bash[2393]: INFO: __main__: Skipping fedora-29/external_ca_TestExternalCAdirsrvStop of #6: waiting for dependencies
jun 18 16:16:23 reproducer-3130.lab.cerberus.domain bash[2393]: INFO: __main__: Skipping fedora-29/external_ca_TestExternalCAInvalidCert of #6: waiting for dependencies
jun 18 16:16:23 reproducer-3130.lab.cerberus.domain bash[2393]: INFO: __main__: Skipping fedora-29/external_ca_TestMultipleExternalCA of #6: waiting for dependencies
jun 18 16:16:23 reproducer-3130.lab.cerberus.domain bash[2393]: INFO: __main__: Skipping fedora-29/test_ntp_options of #6: not enough resources
jun 18 16:16:23 reproducer-3130.lab.cerberus.domain bash[2393]: INFO: __main__: Skipping fedora-29/test_pkinit_manage of #6: waiting for dependencies
jun 18 16:16:23 reproducer-3130.lab.cerberus.domain bash[2393]: INFO: __main__: Skipping fedora-29/test_automount_locations of #6: waiting for dependencies
jun 18 16:16:23 reproducer-3130.lab.cerberus.domain bash[2393]: INFO: __main__: Skipping fedora-29/mask of #6: waiting for dependencies
[root@reproducer-3130 ~]# journalctl -f --unit=prci.service
-- Logs begin at Wed 2019-05-29 15:53:35 -03. --
jun 18 16:16:54 reproducer-3130.lab.cerberus.domain bash[2393]: INFO: __main__: Skipping fedora-29/test_webui_users of #6: waiting for dependencies
jun 18 16:16:54 reproducer-3130.lab.cerberus.domain bash[2393]: INFO: __main__: Skipping fedora-29/customized_ds_config_install of #6: waiting for dependencies
jun 18 16:16:54 reproducer-3130.lab.cerberus.domain bash[2393]: INFO: __main__: Skipping fedora-29/dns_locations of #6: not enough resources
jun 18 16:16:54 reproducer-3130.lab.cerberus.domain bash[2393]: INFO: __main__: Skipping fedora-29/external_ca_TestExternalCAdirsrvStop of #6: waiting for dependencies
jun 18 16:16:54 reproducer-3130.lab.cerberus.domain bash[2393]: INFO: __main__: Skipping fedora-29/external_ca_TestExternalCAInvalidCert of #6: waiting for dependencies
jun 18 16:16:54 reproducer-3130.lab.cerberus.domain bash[2393]: INFO: __main__: Skipping fedora-29/external_ca_TestMultipleExternalCA of #6: waiting for dependencies
jun 18 16:16:54 reproducer-3130.lab.cerberus.domain bash[2393]: INFO: __main__: Skipping fedora-29/test_ntp_options of #6: not enough resources
jun 18 16:16:54 reproducer-3130.lab.cerberus.domain bash[2393]: INFO: __main__: Skipping fedora-29/test_pkinit_manage of #6: waiting for dependencies
jun 18 16:16:54 reproducer-3130.lab.cerberus.domain bash[2393]: INFO: __main__: Skipping fedora-29/test_automount_locations of #6: waiting for dependencies
jun 18 16:16:54 reproducer-3130.lab.cerberus.domain bash[2393]: INFO: __main__: Skipping fedora-29/mask of #6: waiting for dependencies
jun 18 16:17:25 reproducer-3130.lab.cerberus.domain bash[2393]: INFO: __main__: Attempting to lock a task fedora-29/build for PR#6.
jun 18 16:17:55 reproducer-3130.lab.cerberus.domain bash[2393]: INFO: __main__: fedora-29/build PR#6 is successfully locked.
jun 18 16:17:55 reproducer-3130.lab.cerberus.domain bash[2393]: INFO: __main__: Available resources: 6 CPU, 3240.78125MB
jun 18 16:17:55 reproducer-3130.lab.cerberus.domain bash[2393]: INFO: root: Initializing job c6d09a26-91fd-11e9-8ac8-001c42026ead
jun 18 16:17:55 reproducer-3130.lab.cerberus.domain bash[2393]: INFO: root: Executing: VagrantBoxDownload
jun 18 16:17:55 reproducer-3130.lab.cerberus.domain bash[2393]: INFO: root: Executing: VagrantUp
jun 18 16:17:55 reproducer-3130.lab.cerberus.domain bash[2393]: INFO: root: Executing: Process "vagrant up --no-provision --parallel"
jun 18 16:17:56 reproducer-3130.lab.cerberus.domain bash[2393]: DEBUG: root: Bringing machine 'builder' up with 'libvirt' provider...
jun 18 16:17:57 reproducer-3130.lab.cerberus.domain bash[2393]: DEBUG: root: ==> builder: Creating image (snapshot of base box volume).
jun 18 16:17:57 reproducer-3130.lab.cerberus.domain bash[2393]: DEBUG: root: ==> builder: Creating domain with the following settings...
jun 18 16:17:57 reproducer-3130.lab.cerberus.domain bash[2393]: DEBUG: root: ==> builder:  -- Name:              c6d09a26-91fd-11e9-8ac8-001c42026ead_builder
jun 18 16:17:57 reproducer-3130.lab.cerberus.domain bash[2393]: DEBUG: root: ==> builder:  -- Domain type:       kvm
jun 18 16:17:57 reproducer-3130.lab.cerberus.domain bash[2393]: DEBUG: root: ==> builder:  -- Cpus:              2
jun 18 16:17:57 reproducer-3130.lab.cerberus.domain bash[2393]: DEBUG: root: ==> builder:  -- Feature:           acpi
jun 18 16:17:57 reproducer-3130.lab.cerberus.domain bash[2393]: DEBUG: root: ==> builder:  -- Feature:           apic
jun 18 16:17:57 reproducer-3130.lab.cerberus.domain bash[2393]: DEBUG: root: ==> builder:  -- Feature:           pae
jun 18 16:17:57 reproducer-3130.lab.cerberus.domain bash[2393]: DEBUG: root: ==> builder:  -- Memory:            3800M
jun 18 16:17:57 reproducer-3130.lab.cerberus.domain bash[2393]: DEBUG: root: ==> builder:  -- Management MAC:
jun 18 16:17:57 reproducer-3130.lab.cerberus.domain bash[2393]: DEBUG: root: ==> builder:  -- Loader:
jun 18 16:17:57 reproducer-3130.lab.cerberus.domain bash[2393]: DEBUG: root: ==> builder:  -- Base box:          freeipa/ci-master-f29
jun 18 16:17:57 reproducer-3130.lab.cerberus.domain bash[2393]: DEBUG: root: ==> builder:  -- Storage pool:      default
jun 18 16:17:57 reproducer-3130.lab.cerberus.domain bash[2393]: DEBUG: root: ==> builder:  -- Image:             /var/lib/libvirt/images/c6d09a26-91fd-11e9-8ac8-001c42026ead_builder.img (41G)
jun 18 16:17:57 reproducer-3130.lab.cerberus.domain bash[2393]: DEBUG: root: ==> builder:  -- Volume Cache:      unsafe
jun 18 16:17:57 reproducer-3130.lab.cerberus.domain bash[2393]: DEBUG: root: ==> builder:  -- Kernel:
jun 18 16:17:57 reproducer-3130.lab.cerberus.domain bash[2393]: DEBUG: root: ==> builder:  -- Initrd:
jun 18 16:17:57 reproducer-3130.lab.cerberus.domain bash[2393]: DEBUG: root: ==> builder:  -- Graphics Type:     none
jun 18 16:17:57 reproducer-3130.lab.cerberus.domain bash[2393]: DEBUG: root: ==> builder:  -- Graphics Port:     5900
jun 18 16:17:57 reproducer-3130.lab.cerberus.domain bash[2393]: DEBUG: root: ==> builder:  -- Graphics IP:       127.0.0.1
jun 18 16:17:57 reproducer-3130.lab.cerberus.domain bash[2393]: DEBUG: root: ==> builder:  -- Graphics Password: Not defined
jun 18 16:17:57 reproducer-3130.lab.cerberus.domain bash[2393]: DEBUG: root: ==> builder:  -- Video Type:        cirrus
jun 18 16:17:57 reproducer-3130.lab.cerberus.domain bash[2393]: DEBUG: root: ==> builder:  -- Video VRAM:        9216
jun 18 16:17:57 reproducer-3130.lab.cerberus.domain bash[2393]: DEBUG: root: ==> builder:  -- Sound Type:
jun 18 16:17:57 reproducer-3130.lab.cerberus.domain bash[2393]: DEBUG: root: ==> builder:  -- Keymap:            en-us
jun 18 16:17:57 reproducer-3130.lab.cerberus.domain bash[2393]: DEBUG: root: ==> builder:  -- TPM Path:
jun 18 16:17:57 reproducer-3130.lab.cerberus.domain bash[2393]: DEBUG: root: ==> builder:  -- INPUT:             type=mouse, bus=ps2
jun 18 16:17:58 reproducer-3130.lab.cerberus.domain bash[2393]: DEBUG: root: ==> builder: Creating shared folders metadata...
jun 18 16:17:58 reproducer-3130.lab.cerberus.domain bash[2393]: DEBUG: root: ==> builder: Starting domain.
jun 18 16:17:58 reproducer-3130.lab.cerberus.domain bash[2393]: DEBUG: root: ==> builder: Waiting for domain to get an IP address...
^C
[root@reproducer-3130 ~]# pgrep vagrant
[root@reproducer-3130 ~]# ps aux | grep -i vagrant
dnsmasq   1449  0.0  0.0  24296  2120 ?        S    14:54   0:00 /usr/sbin/dnsmasq --conf-file=/var/lib/libvirt/dnsmasq/vagrant-libvirt.conf --leasefile-ro --dhcp-script=/usr/libexec/libvirt_leaseshelper
root      1450  0.0  0.0  24268   408 ?        S    14:54   0:00 /usr/sbin/dnsmasq --conf-file=/var/lib/libvirt/dnsmasq/vagrant-libvirt.conf --leasefile-ro --dhcp-script=/usr/libexec/libvirt_leaseshelper
root      2532  5.8  1.1 434472 92608 ?        Ssl  16:17   0:01 /usr/bin/ruby-mri /usr/share/vagrant/gems/bin/vagrant up --no-provision --parallel
root      2680  0.0  0.0 215744   960 pts/1    S+   16:18   0:00 grep --color=auto -i vagrant
[root@reproducer-3130 ~]# vagrant global-status
id       name    provider state     directory
--------------------------------------------------------------------------------------------------
e5408d7  builder libvirt preparing /root/freeipa-pr-ci/jobs/c6d09a26-91fd-11e9-8ac8-001c42026ead

The above shows information about all known Vagrant environments
on this machine. This data is cached and may not be completely
up-to-date (use "vagrant global-status --prune" to prune invalid
entries). To interact with any of the machines, you can go to that
directory and run Vagrant, or you can use the ID directly with
Vagrant commands from any directory. For example:
"vagrant destroy 1a2b3c4d"
```